packages/preact/src/index.ts

```diff
@@ -332,7 +332,7 @@ Component.prototype.shouldComponentUpdate = function (
 };
 
 export function useSignal<T>(value: T) {
-	return useMemo(() => signal<T>(value), []);
```

Comment:
I think that one important detail to note is that the `useRef` version is passing in `signal<T>(value)` as compared to `() => signal<T>(value)` for `useMemo`.

This means that whenever `useSignal` is reran from say state changes, **a new signal obj is created**, even though the new signal objs are effectively ignored.

Overall, I do think that the orginal `useMemo` is actually the better version.

Reply:
@billybimbob Hmm good point. That said, the `useMemo`-version creates a new method (`() => signal<T>(value)`) every time `useSignal` gets (re)called? (I do assume creating a signal is more work than creating a function that's never called?)

Reply:
That's also a good point about the methods.

I guess really with either form, an object is being created in some form, either a signal object or a function.

I personally don't know how much overhead is added when creating an arrow function as compared to any other objects, but at least in my opinion, the *semantics* of creating a function that is ignored makes more sense than an actual signal value that is ignored.

Reply:
Yeah, and in this case `useRef` actually uses `useMemo` under the hood [link](https://github.com/preactjs/preact/blob/master/hooks/src/index.js#L265), so I guess this change doesn't make sense. At least not for preact (I think react's implementation of `useRef` is different, so maybe the switch from `useMemo` still makes sense there).